### PR TITLE
Working factorial. **Only works with a right operand. Right operand i…

### LINKED ISSUE #11: Feature Advanced mode
Currently, there is no option for advanced/extended mode for additional operations. PyCalc is limited to addition, subtraction, multiplication, division, and decimal operations only. The goal is to extend to more functionalities by **toggling** an *advanced* or *extended* button. The following features may be taken into consideration:
1. - [x] brackets-open/close `(` `)` 
     - From cec12766368e826250a69e65f68b9919e2c27700 brackets are entred using the keyboard in Basic mode.
     - Added in commit be65811f225928acbf918ac192d9b69add99d0bd.
2. - [x] power `^`
     - Added in commit 05515f9568ec4d4d74c001e5532a581e7b08f07a.
3. - [ ] factorial `!`
     - Logic to be added.
4. - [ ] logarithm `log`:  -> provide option for changing base 10
     - Logic to be added.
5. - [ ] trigonometric operations `sin` `cos` `tan`  -> provide option for changing radians to degrees
     - Logic to be added.

Discussion on implementation ideas is welcome :smile: .

### PR DESCRIPTION
**Pull request by nathdilla**
…s disregarded.

### CODE REVIEW

File: core_logic.py [hunk 1]
```diff
@@ -19,13 +19,14 @@ class Calculate():
...
     _get_value = {
+            'x!': lambda x, y: Calculate.factorial(None,int(x),int(y)),
             '^': lambda x, y: x**y,
             '/': lambda x, y: x/y,
             '*': lambda x, y: x*y,
             '+': lambda x, y: x+y,
             '-': lambda x, y: x-y
             }
-    __operators: list = ['(', '^', '/', '*', '+', '-']
+    __operators: list = ['(', '^', '/', '*', '+', '-', 'x!']
```
Comment (maddypie) [inline core_logic.py:29]: Comply to BODMAS rule. Insert the factorial in index 1.

File: core_logic.py [hunk 2]
```diff
@@ -19,13 +19,14 @@ class Calculate():
 
     ans: str
     _get_value = {
+            'x!': lambda x, y: Calculate.factorial(None,int(x),int(y)),
```
Comment (maddypie) [inline core_logic.py:22]: Using `math.factorial` will be a better way to do it.

File: core_logic.py [hunk 3]
```diff
@@ -129,3 +133,12 @@ def __call_partial_calculate(self, operator):
     def __create_op_gen(self, operator: list):
         generator = (g for g in operator if g in self.expr_as_list)
         return generator
+
+    def factorial(self, num, num2):
+        print(self, num, num2)
+        result = 1
+        for i in range(1, num + 1):
+            result = result * i
+        print(result)
```
Comment (maddypie) [inline core_logic.py:142]: Remove debug-print statements.

Comment (maddypie) [inline core_logic.py:142]: Can be replaced with `math.factorial`.

File: core_logic.py [hunk 4]
```diff
@@ -61,6 +62,9 @@ def calculate(self) -> str:
                     new_instance.calculate()
                     ]
 
+        fact_op = self.__operators[6]
+        if any(fact_op):
```
Comment (maddypie) [inline core_logic.py:66]: Missed to form a *generator* for `fact_op`. See line 69 for reference.

File: storage.py [hunk 5]
```diff
@@ -14,7 +14,7 @@ class Storage:
         Stores argument `character` if logic is met
     """
 
-    __operators: list = ['^', '/', '*', '+', '-']
+    __operators: list = ['x!','^', '/', '*', '+', '-']
```
Comment (maddypie) [inline storage.py:17]: Don't mix operand `x` and operator `x`. Here `__operator` list contains only operators.

**Review by tsadarsh — CHANGES_REQUESTED** [review]
You did a great job. :+1:  
Try to make as many changes/corrections yourself. I will also be working on this code.

**Comment by maddypie** [comment]
It is **awesome of you** to contribute to this project. I really appreciate your work. You have made some notable lead to advanced-mode-factorial feature. :smile: :+1: 

As to your PR:
 1. Did you run the program from your side to check if it is working. The code you have added breaks the earlier code (view unittest-`checks` in this PR, it fails). If you need to, try refactoring the original code to accomodate your ideas/implementations.
 2.  Do add *unittest* to support your code. Adding unittests are simple. 
 3. Stick to [PEP8](https://www.python.org/dev/peps/pep-0008/).
 4. Add/Edit documentation whenever possible.

Try your best to get the `checks` passing. I will also be editting this PR to get it working. I have reviewed your PR, please go throught it also.  
Please feel free to ping this thread for any clarifications.

**Comment by maddypie** [comment]
Please have a look at branch **unary** for the changes taking place.

**Comment by tsadarsh** [comment]
Pymanujan v2.0 is finally released :partying_face:!!
Many new operators are added. Factorial is all good now.
Thank you for participating :smile:

**Pull request closed without merging**